nowledge-mem-codex-plugin/hooks/nmem-context.py

```
@@ -168,10 +243,19 @@ def main(payload: dict[str, Any] | None = None) -> int:
-            if hook_event_name == "UserPromptSubmit"
-            else ROUTING_GUIDANCE.strip()
-        )
+        if hook_event_name == "UserPromptSubmit":
+            guidance = PROMPT_ROUTING_GUIDANCE
+        elif hook_event_name == "SubagentStart":
+            agent_type = str(hook_payload.get("agent_type") or "default").strip()
+            if (
+                agent_type == "explorer"
+                and agent_type not in _subagent_context_types()
+            ):
+                raise SystemExit(0)
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)